### ISSUE TITLE
dvc>=2.7.3 is incompatible with tensorflow because of typing-extensions

### ISSUE THREAD

**Issue by gcoter** [iterative/dvc#6731]
Hello,

It is not really a feature request but not really a bug neither. Feel free to tell me if I should post this elsewhere.

I work on a project with dvc and tensorflow. I noticed that I did not have the last version (2.7.2 instead of 2.7.4) so I tried to upgrade.

I use Poetry to manage the dependencies of the project. It is quite good to handle incompatibilities. When I ask it to install dvc, it chooses version 2.7.2. When I force it to install version 2.7.4, I get this incompatibility error:

```
$ poetry add "dvc[ssh]==2.7.4"

Updating dependencies
Resolving dependencies... (0.3s)

  SolverProblemError

  Because no versions of tensorflow match >2.6.0,<3.0.0
   and tensorflow (2.6.0) depends on typing-extensions (>=3.7.4,<3.8.0), tensorflow (>=2.6.0,<3.0.0) requires typing-extensions (>=3.7.4,<3.8.0).
  And because dvc (2.7.4) depends on typing-extensions (>=3.10.0.2), tensorflow (>=2.6.0,<3.0.0) is incompatible with dvc (2.7.4).
  So, because music-generation-v2 depends on both dvc (2.7.4) and tensorflow (^2.6.0), version solving failed.
```

In other words, tensorflow requires a version of typing-extensions which is lower than the one required by dvc (since version 2.7.3). For now, I could not find a recent tensorflow version which would be compatible with this requirement.

I found that the requirement was modified in this commit on your side: https://github.com/iterative/dvc/commit/ef017befbe0257173304394a97b8b42cfca3cc52#diff-8a5e2e65dfa49c67096312e68e6620b20055b720b11bcdf44aa003ff02103af2

So my question is: would it be possible to relax this requirement on typing-extensions?

**Comment by efiop** [comment]
Looks like we could exclude 3.10.0.1 and relax the rest to be >= 3.7.4 CC @skshetry wdyt?

**Comment by skshetry** [comment]
@efiop, we can just set it to `>=3.7.4`. On other installations, it'll install the latest version anyway. 

@gcoter PRs are welcome.

**Comment by gcoter** [comment]
@skshetry I will make a PR then :slightly_smiling_face:
